### PR TITLE
fix: [PL-31125]: check user group deletion based on user group managed

### PR DESCRIPTION
**Pull request by tiwaritejas**
### Summary

When deleting a user from the user group, there should be a check for whether the user group is scim managed or not. Currently, the check is run for whether the user is scim managed, which is incorrect as it prevents the scim user from being deleted from user groups which are not externally managed.

Testing:
Locally tested the UI changes for both cases- deleting scim user from managed and non-managed user groups. UI renders as expected.

#### Screenshots

Screenshot 2023-02-01 at 11 45 22 AM

Screenshot 2023-02-01 at 11 45 06 AM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/20-rbac/pages/UserGroupDetails/views/MemberList.tsx [hunk 1]
```diff
@@ -202,8 +215,11 @@ const MemberList: React.FC<MemberListProps> = ({ ssoLinked, userGroupInherited,
     }
   })
 
-  const users = useMemo(() => data?.data?.content, [data?.data])
+  const getColumnMenu = (...args: any) => {
```
Comment (abhinavrastogi-harness) [inline src/modules/20-rbac/pages/UserGroupDetails/views/MemberList.tsx:218]: why do we need this `get` function? you can directly render `<RenderColumnMenu />` on line 244 below

**Comment by tiwaritejas** [comment]
retrigger jest

**Comment by tiwaritejas** [comment]
retrigger titlecheck

**Review by abhinavrastogi-harness — APPROVED** [review]

**Comment by tiwaritejas** [comment]
retrigger cypress-pipeline

**Review by swarajcheguri-harness — APPROVED** [review]

**Pull request merged**